ProjectName/src/screens/PaymentDetails/PaymentDetailsForm/components/customAmountInput/CustomAmountInput.tsx

```diff
@@ -0,0 +1,53 @@
+import React, {FC} from 'react';
+import {Controller} from 'react-hook-form';
+import {View, Text, TextInput, Image} from 'react-native';
+
+import {DECIMAL_REGEX} from './custom-amount-input.constants';
+import {styles} from './custom-amount-input.styles';
+import {CustomMessageInputTypes} from '../customMessageInput/custom-message-input.types';
+
+const CustomAmountInput: FC<CustomMessageInputTypes> = ({
+  control,
+  name,
+  title,
+  placeholder,
+}) => {
+  return (
+    <View>
+      <Text style={styles.title}>{title}</Text>
+      <Controller
+        control={control}
+        name={name}
+        render={({field: {value, onChange, onBlur}}) => (
+          <View style={styles.container}>
+            <TextInput
+              keyboardType="decimal-pad"
+              value={value}
+              onChangeText={text => {
+                if (DECIMAL_REGEX.test(text)) {
+                  const formattedValue = text;
+                  onChange(formattedValue);
+                }
+              }}
+              onBlur={onBlur}
+              placeholder={placeholder}
+              maxLength={12}
```

Comment:
It's cool you using built-in functionality for such purposes, but I would suggest to make basic component more generic and accept only harly required parameters. Just imagine the other day we would need to reuse amount input, but with different `maxLength`. It would be hard to achieve it it's hardcoded inside the component itself.